[UI] Switch the ui theme to dark mode

ChengYanJin opened the pull request:
**Component**:
ui

**Context**: 
Switch the UI to the dark mode to be aligned with the new design 

**Summary**:

- Switch the theme to the dark mode
- As well as the dex login in page
- Update the lateral navbar with the new design
- Update the font-face with `Lato`
- Merge some colors in color palette
  textSecondary (#A8B5C1) & textTertiary (#A7B6C3) => keep text Secondary #A8B5C1
  info button color (#027AFF) & Secondary color is blue (#037AFF) => > Keep #037AFF which 
  should be the color of the most important action button, and the selection feedback color.

**Acceptance criteria**:
Please find the screenshots: 

![image](https://user-images.githubusercontent.com/18453133/81907130-917adc00-95c7-11ea-9050-0eb052613fb7.png)
 
![image](https://user-images.githubusercontent.com/18453133/81906982-62fd0100-95c7-11ea-9897-a17e00fa9bd5.png)

![image](https://user-images.githubusercontent.com/18453133/81907463-19f97c80-95c8-11ea-9332-2f915eb23295.png)

---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: #2528 

<!-- If you want to refer to an issue while not closing it, use:

See: #ISSUE_NUMBER

-->

ChengYanJin committed:
ui/theme: apply the new color palette

ChengYanJin committed:
ui/components: modify the components to adapt to the new color palette

ChengYanJin committed:
ui/assets: Add the Lato font assets & delete the previous fonts

Refs: #2427

ChengYanJin committed:
ui/fonts: Add the lato into the font-face

Refs: #2427

ChengYanJin committed:
ui/layout: Add toggle in sidebar

Refs: #2486

ChengYanJin committed:
ui/package: Update with latest core-ui version

Refs: #2528

ChengYanJin committed:
ui/theme: Switch the theme & dex login to dark mode

Refs: #2528

bert-e commented:
# Hello chengyanjin,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.6/improvement/2528-switch-the-ui-theme-to-dark-mode` with contents from `improvement/2528-switch-the-ui-theme-to-dark-mode`
and `development/2.6`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.6/improvement/2528-switch-the-ui-theme-to-dark-mode origin/development/2.6
 $ git merge origin/improvement/2528-switch-the-ui-theme-to-dark-mode
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.6/improvement/2528-switch-the-ui-theme-to-dark-mode
```

ChengYanJin committed:
ui/color-palette: Merge the color palette

- textSecondary (#A8B5C1) & textTertiary (#A7B6C3) => keep text Secondary #A8B5C1
- info button color (#027AFF) & Secondary color is blue (#037AFF) => > Keep #037AFF which should be the color of the most important action button, and the selection feedback color.

Refs: #2528

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @nootal

* @alexandre-allard-scality

* @lucieleonard

* @sayf-eddine-scality

* @slaperche-scality

* @nicolast

* @gdemonet

* @carlito-scality

* @thomasdanan

* @jbertran

* @teddyandrieux

* @cmonfort

* @monpote

bert-e commented:
# Conflict




A conflict has been raised during the update of
integration branch `w/2.6/improvement/2528-switch-the-ui-theme-to-dark-mode` with contents from `improvement/2528-switch-the-ui-theme-to-dark-mode`
and `development/2.6`.

Please resolve the conflict on **the integration branch** (`w/2.6/improvement/2528-switch-the-ui-theme-to-dark-mode`).


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout w/2.6/improvement/2528-switch-the-ui-theme-to-dark-mode
 $ git pull  # or "git reset --hard origin/w/2.6/improvement/2528-switch-the-ui-theme-to-dark-mode"
 $ git merge origin/development/2.6
 $ # <intense conflict resolution>
 $ git commit
 $ git merge origin/improvement/2528-switch-the-ui-theme-to-dark-mode
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.6/improvement/2528-switch-the-ui-theme-to-dark-mode
```

bert-e commented:
# History mismatch



Merge commit #be9bd3e5a28bab58564d76e83a59f393586f62e8 on the integration branch
`w/2.6/improvement/2528-switch-the-ui-theme-to-dark-mode` is merging a branch which is neither the current
branch `improvement/2528-switch-the-ui-theme-to-dark-mode` nor the development branch
`development/2.6`.

It is likely due to a rebase of the branch `improvement/2528-switch-the-ui-theme-to-dark-mode` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

gdemonet commented:
@ChengYanJin we should be able to merge this PR now that 2.5.1 is released :)

bert-e commented:
# Branches have diverged



This pull request's source branch `improvement/2528-switch-the-ui-theme-to-dark-mode` has diverged from
`development/2.5` by more than 50 commits.

To avoid any integration risks, please re-synchronize them using one of the
following solutions:

* Merge `origin/development/2.5` into `improvement/2528-switch-the-ui-theme-to-dark-mode`
* Rebase `improvement/2528-switch-the-ui-theme-to-dark-mode` onto `origin/development/2.5`

Note: If you choose to rebase, you may have to ask me to rebuild
integration branches using the `reset` command.

bert-e commented:
# History mismatch



Merge commit #be9bd3e5a28bab58564d76e83a59f393586f62e8 on the integration branch
`w/2.6/improvement/2528-switch-the-ui-theme-to-dark-mode` is merging a branch which is neither the current
branch `improvement/2528-switch-the-ui-theme-to-dark-mode` nor the development branch
`development/2.6`.

It is likely due to a rebase of the branch `improvement/2528-switch-the-ui-theme-to-dark-mode` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

ChengYanJin committed:
ui/utils: Add util functions used by volume page

- allSizeUnitsToBytes: Convert all the units into bytes
- bytesToSize: convert the bytes to the size with unit
- formatDate: Formate the date to YYYY/MM/DD
- computeVolumeCondition: Compute the condition of the volume based ont the current status and isbound
- jointDataPointBaseonTimeSeries: Sort the data points according to the time series
- addMissingDataPoint: Add the missing data point with `null` value which caused by the downtime of SC

ChengYanJin committed:
ui/volumepage: Volume list table & Rightside panel

Refs: #2506

ChengYanJin committed:
ui/volumes: monitoring saga

ChengYanJin committed:
ui/volumes: Active Alert Component

Refs: #2508

ChengYanJin committed:
ui/volumes: Volume Detail Component

ChengYanJin committed:
ui/volumes: Performance graph

- Manually add the missing data points with `null` value, then vega-lite could address it without interpolating the absence of data
- Remove the max/min label for to keep the consistent for all the graphs

Refs: #2507

ChengYanJin committed:
ui/volumes: Add the AlertManager service

ChengYanJin committed:
ui/translation: Update the translation for Volume Page

ChengYanJin committed:
ui/layout: Update the layout

ChengYanJin committed:
ui/nodes: Add the instance ip inside the Node object

- When we do the query for prometheus, we want to specify the instance ip

ChengYanJin committed:
ui/package: Install react-table $ update the Core-ui version

ChengYanJin committed:
ui/layout: Deactivate the access to volume page from the sidebar

Since we only can access the volume page through node list, which means the volumes are filtered by the specific node
We will implement the globle volume list later which is the whole volumes.

Refs: #2505

ChengYanJin commented:
rename the branch
